Tests/Tensor/test_Tensor_conj.cpp

```diff
@@ -1,71 +1,78 @@
 #include "../test.hpp"
-#include "NSL.hpp"
 
-template<NSL::Concept::isComplex Type, NSL::Concept::isIntegral ... SizeTypes>
-void test_conj_transpose(SizeTypes ... Ns);
+NSL_TEST_CASE("Tensor Transpose", "[Tensor, Transpose]"){
+    NSL::size_t size = GENERATE(1,2,4,8,32,64);
 
-COMPLEX_NSL_TEST_CASE("Tensor 1D Complex Conjugate", "[Tensor,1D,Complex Conjugate]"){
-    NSL::size_t size0 = GENERATE(1,8,32);
-    test_conj_transpose<TestType>(size0);
-}
+    NSL::Tensor<TestType> A(size,size); 
+    if constexpr(NSL::Concept::isFloatingPoint<TestType>){
+        A.rand();
+    } else if(NSL::Concept::isType<TestType,bool>){
+        A(NSL::Slice(0,size,2)) = true;
+    }  else{ 
+        A.randint(0,size);
+    }
+    NSL::Tensor<TestType> Abak(A,true);
+    
+    A.transpose();
 
-COMPLEX_NSL_TEST_CASE("Tensor 2D Complex Conjugate", "[Tensor,2D,Complex Conjugate]"){
-    NSL::size_t size0 = GENERATE(1,8,32);
-    NSL::size_t size1 = GENERATE(1,8,32);
-    test_conj_transpose<TestType>(size0,size1);
-}
+    for(NSL::size_t i = 0; i < size; ++i){
+        for(NSL::size_t j = 0; j < size; ++j){
+            REQUIRE(A(i,j) == Abak(j,i));
+        }
+    }
+
+    // restore default
+    A = Abak;
 
-COMPLEX_NSL_TEST_CASE("Tensor 3D Complex Conjugate", "[Tensor,3D,Complex Conjugate]"){
-    NSL::size_t size0 = GENERATE(1,8,32);
-    NSL::size_t size1 = GENERATE(1,8,32);
-    NSL::size_t size2 = GENERATE(1,8,32);
-    test_conj_transpose<TestType>(size0,size1,size2);
+    A.transpose(0,1);
+
+    for(NSL::size_t i = 0; i < size; ++i){
+        for(NSL::size_t j = 0; j < size; ++j){
+            REQUIRE(A(i,j) == Abak(j,i));
+        }
+    }
 }
```

Comment:
We used to test the conj/transpose/adjoint on tensors of dimension > 2 (there was 2D, 3D, and [below,] 4D).  This is an important test.  Why is it no longer tested?

Reply:
The previous test was faulty! I should reinsert a test that does a batch transpose etc.

Reply:
Addressed in f6f3373888a7765d5bec8fceafe769a41e4b06bc